pkg/kubelet/kubelet_node_status.go

```diff
@@ -315,7 +315,7 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 // It synchronizes node status to master, registering the kubelet first if
 // necessary.
 func (kl *Kubelet) syncNodeStatus() {
-	if kl.kubeClient == nil {
+	if kl.kubeClient == nil || kl.standaloneMode {
```

Comment:
It's theoretically possible for a wrapper around the Kubelet to pass in a client as part of the KubeletDeps while also triggering standalone mode via the values on KubeletServer. I'd rather just guard against that.